payjoin-cli/src/cli/mod.rs

```diff
@@ -0,0 +1,135 @@
+use std::path::PathBuf;
+
+use clap::{value_parser, Parser, Subcommand};
+use payjoin::bitcoin::amount::ParseAmountError;
+use payjoin::bitcoin::{Amount, FeeRate};
+use serde::Deserialize;
+use url::Url;
+
+#[derive(Debug, Clone, Deserialize, Parser)]
+pub struct Flags {
+    #[arg(long = "bip77", help = "Use BIP77 (v2) protocol (default)", action = clap::ArgAction::SetTrue)]
+    pub bip77: Option<bool>,
+    #[arg(long = "bip78", help = "Use BIP78 (v1) protocol", action = clap::ArgAction::SetTrue)]
+    pub bip78: Option<bool>,
+}
+
+#[derive(Debug, Parser)]
+#[command(
+    version = env!("CARGO_PKG_VERSION"),
+    about = "Payjoin - bitcoin scaling, savings, and privacy by default",
+    long_about = None,
+    subcommand_required = true
+)]
+pub struct Cli {
+    #[command(flatten)]
+    pub flags: Flags,
+
+    #[command(subcommand)]
+    pub command: Commands,
+
+    #[arg(long, short = 'd', help = "Sets a custom database path")]
```

Comment:
A nit for the commit log and title of the PR is that these are [procedural aka "proc" macros](https://doc.rust-lang.org/reference/procedural-macros.html), not "derive" macros. `derive` is just a common example of proc macro usage.

Reply:
git commit --amend it please